src/main/automation/BaseActionService.ts

```
@@ -0,0 +1,243 @@
+  }
+
+  private calculateAttributeScore(
+    node: any,
+    targetAttributes: Record<string, string>
+  ): number {
+    if (!node.properties || node.properties.length === 0) {
+      return 0;
+    }
+
+    let matchedCount = 0;
+    let totalCount = Object.keys(targetAttributes).length;
```

ESLint

[error] 'totalCount' is never reassigned. Use 'const' instead.

(prefer-const)